sagemaker_processing/spark_distributed_data_processing/sagemaker-spark-processing.ipynb

```diff
@@ -0,0 +1,562 @@
+    "# Upload the raw input dataset to S3\n",
+    "timestamp_prefix = strftime(\"%Y-%m-%d-%H-%M-%S\", gmtime())\n",
+    "prefix = 'sagemaker/spark-preprocess-demo/' + timestamp_prefix\n",
+    "input_prefix_sales = prefix + '/input/sales'\n",
+    "output_prefix_sales = prefix + '/output/sales'\n",
+    "input_s3_uri = 's3://' + bucket + '/' + input_prefix_sales\n",
+    "output_s3_uri = 's3://' + bucket + '/' + output_prefix_sales\n",
+    "\n",
+    "sagemaker_session.upload_data(path='./data/data.jsonl', bucket=bucket, key_prefix=input_prefix_sales)\n",
+    "\n",
+    "spark_processor = SparkJarProcessor(\n",
+    "    base_job_name=\"sm-spark-java\",\n",
+    "    framework_version=\"2.4\",\n",
+    "    role=role,\n",
+    "    instance_count=2,\n",
+    "    instance_type=\"ml.m5.xlarge\",\n",
+    "    max_runtime_in_seconds=1200,\n",
+    ")\n",
+    "\n",
+    "spark_processor.run(\n",
+    "    submit_app=\"./code/spark-test-app.jar\",\n",
+    "    submit_class=\"com.amazonaws.sagemaker.spark.test.HelloJavaSparkApp\",\n",
+    "    arguments=[\"--input\", input_s3_uri, \"--output\", output_s3_uri],\n",
+    "    logs=False\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Example 4: Specifying additional Spark configuration\n",
+    "\n",
+    "Overriding Spark configuration is crucial for a number of tasks such as tuning your Spark application or configuring the hive metastore. Using the SageMaker Python SDK, you can easily override Spark/Hive/Hadoop configuration.\n",
+    "\n",
+    "An example usage would be overriding Spark executor memory/cores as demonstrated in the next example.\n",
+    "\n",
+    "For more information on configuring your Spark application, see the EMR documentation on [Configuring Applications](https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-configure-apps.html)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Upload the raw input dataset to a unique S3 location\n",
+    "timestamp_prefix = strftime(\"%Y-%m-%d-%H-%M-%S\", gmtime())\n",
+    "prefix = 'sagemaker/spark-preprocess-demo/' + timestamp_prefix\n",
+    "input_prefix_abalone = prefix + '/input/raw/abalone'\n",
+    "input_preprocessed_prefix_abalone = prefix + '/input/preprocessed/abalone'\n",
+    "\n",
+    "sagemaker_session.upload_data(path='./data/abalone.csv', bucket=bucket, key_prefix=input_prefix_abalone)\n",
+    "\n",
+    "spark_processor = PySparkProcessor(\n",
+    "    base_job_name=\"sm-spark\",\n",
+    "    framework_version=\"2.4\",\n",
+    "    role=role,\n",
+    "    instance_count=2,\n",
+    "    instance_type=\"ml.m5.xlarge\",\n",
+    "    max_runtime_in_seconds=1200,\n",
+    ")\n",
+    "\n",
+    "configuration = [{\n",
+    "    \"Classification\": \"spark-defaults\",\n",
+    "    \"Properties\": {\"spark.executor.memory\": \"2g\", \"spark.executor.cores\": \"1\"},\n",
```

Comment:
Wondering what kind of interesting example we can provide here.

Reply:
Yea, we struggled to find a better/more realistic example here. This isn't a realistic example where tuning executor mem/cores is required - but effectively demonstrates how a customer would go about doing it.

Reply:
:+1: